src/H5T.c

```diff
@@ -5157,80 +5223,40 @@ H5T__path_find_real(const H5T_t *src, const H5T_t *dst, const char *name, H5T_co
     if (H5CX_pushed() && (H5CX_get_dt_conv_cb(&tmp_ctx.u.init.cb_struct) < 0))
         HGOTO_ERROR(H5E_DATATYPE, H5E_CANTGET, NULL, "unable to get conversion exception callback");
 
-    /*
-     * Make sure the first entry in the table is the no-op conversion path.
-     */
-    if (0 == H5T_g.npaths) {
```

Comment:
Moved the allocation and initialization of the path table into a helper function